pkg/apis/componentconfig/types_pluginargs.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfig
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/descheduler/pkg/api"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RemovePodsViolatingNodeTaintsArgs holds arguments used to configure the RemovePodsViolatingNodeTaints plugin.
+type RemovePodsViolatingNodeTaintsArgs struct {
```

Comment:
This file is used for all the plugin arguments. It's kept under `pkg/apis/componentconfig` as an internal variant of all the plugin arguments. Once the migration is finished, we will define v1alpha2 variant of all the args here so they can be incorporated in the v1alpha2 descheduler policy configuration.

Reply:
@damemi would be great to eventually get rid of the `pkg/api.DeschedulerPolicy` type and integrate the profiles into the `pkg/apis/componentconfig.DeschedulerConfiguration` type. Passing marshalled `DeschedulerConfiguration` as a file through the `--config` flag to pass entire configuration as a single unit. `DeschedulerConfiguration` is unseen from the outside and just complicates the code unnecessarily.

Reply:
Agree that we need to do something with the componentconfig, whether that's leveraging it as the entire config or just dropping it. But having both doesn't add much benefit imo. Maybe reopen https://github.com/kubernetes-sigs/descheduler/issues/431